www/js/app.js

```diff
@@ -1360,6 +1439,7 @@ function archiveReadyCallback (archive) {
 function setLocalArchiveFromFileSelect () {
     setLocalArchiveFromFileList(document.getElementById('archiveFiles').files);
 }
+window.setLocalArchiveFromFileSelect = setLocalArchiveFromFileSelect;
```

Comment:
Did you try simply defining the whole function as a variable: `window.setLocalArchiveFromFileSelect = function () { ... };`? It should still work everywhere as `setLocalArchiveFromFileSelect` (without `window`), so you don't have to redefine it here?

I'm a little concerned that this solution pollutes the global scope and breaks the modular architecture, but I see a previous senior Dev used a similar solution for loading files in Unit tests (and I adopted it when I rewrote the tests for more modern platforms), so I suppose it's OK, and I can't think of a better solution... (!)

Reply:
I don't think there is any other way to do it at least I am not aware of.
I did try to set `setLocalArchiveFromFileSelect` as a global variable but that too didn't work even

Reply:
(just unresolving while you try it)

Reply:
> Maybe just try window.setLocalArchiveFromFileSelect = function () { ... }; for the function def, and see if it works without the extra inserted line here? If it doesn't work, keep what you have. If it works, it might avoid having two functions hanging around that do the same thing (one in app.js and one attached to window).

There is a downside to this `setLocalArchiveFromFileSelect` is referenced in a lot of places and converting it to `window.setLocalArchiveFromFileSelect = function () { ... };` that means everywhere i need to update to 
`setLocalArchiveFromFileSelect` -> `window.setLocalArchiveFromFileSelect` (doesn't seem like a good idea to me)

Reply:
I completely agree, it's just that normally you can access any function in the `window` global scope merely by using the `functionName`, you don't have to use `window.functionName` usually. So it might just work without changing anything else. Try it! If it doesn't, keep what you have.